web/packages/teleport/src/UnifiedResources/ResourceActionButton.test.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { MemoryRouter } from 'react-router';
+
+import { fireEvent, render, screen, waitFor } from 'design/utils/testing';
+
+import { SamlAppActionProvider } from 'teleport/SamlApplications/useSamlAppActions';
+import { App } from 'teleport/services/apps';
+
+import { ResourceActionButton } from './ResourceActionButton';
+
+test('saml app: empty launch url should display <a> link with href value pointed to samlAppSsoUrl', () => {
+  render(
+    <SamlAppActionProvider>
+      <ResourceActionButton resource={resource} />
+    </SamlAppActionProvider>
+  );
+
+  expect(screen.getByRole('link', { name: /Log In/i })).toHaveAttribute(
+    'href',
+    resource.samlAppSsoUrl
+  );
+});
+
+test('saml app: launch url with one item should display <a> link with href value pointed to first launch url item', () => {
```

Comment:
I saw the code for this and thought that it might be easy to break that by mistake, it's good that you added this test.